src/astro/databases/postgres.py

```diff
@@ -70,20 +73,31 @@ def load_pandas_dataframe_to_table(
         :param if_exists: Strategy to be used in case the target table already exists.
         :param chunk_size: Specify the number of rows in each batch to be written at a time.
         """
-        schema = None
         if target_table.metadata and target_table.metadata.schema:
             self.create_schema_if_needed(target_table.metadata.schema)
-            schema = target_table.metadata.schema.lower()
-        source_dataframe.to_sql(
-            target_table.name,
-            schema=schema,  # type: ignore
-            con=self.sqlalchemy_engine,
+
+        # create an empty table
+        db = SQLDatabase(engine=self.sqlalchemy_engine)
+        db.prep_table(
+            source_dataframe,
+            target_table.name.lower(),
+            schema=target_table.metadata.schema,
             if_exists=if_exists,
-            chunksize=chunk_size,
-            method="multi",
             index=False,
         )
 
+        output_buffer = io.StringIO()
+        source_dataframe.to_csv(output_buffer, sep=",", header=True, index=False)
```

Comment:
@dimberman - how do we know the sep="," and header=True are always applied to all CSV? are we in control of how this CSV file is generated?

Reply:
@utkarsharma2 yes we are. The CSV is being generated by pandas (because we first put the data into a dataframe). This makes the user interface much easier because otherwise we'd need to expose all the options.